sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -305,6 +315,27 @@ private[joins] class UnsafeHashedRelation(
   override def read(kryo: Kryo, in: Input): Unit = Utils.tryOrIOException {
     read(() => in.readInt(), () => in.readLong(), in.readBytes)
   }
+
+  override def values(): Iterator[InternalRow] = {
+    if (isLookupAware) {
```

Comment:
shall we always support it? getting all the values seems like a reasonable and general API. Then we don't need the `isLookupAware` flag in this class.

Reply:
@cloud-fan - sounds good. Added a rudimentary trivial implementation of `values()` in `LongHashedRelation`. Also added a unit test in `HashedRelationSuite.scala` - `test("SPARK-32399: test values() method for HashedRelation")` to test behavior for `values()` method.